lib/spack/spack/spec.py

```diff
@@ -1012,9 +1012,14 @@ def __init__(self, spec_like, *dep_like, **kwargs):
         self._hash = other._hash
         self._cmp_key_cache = other._cmp_key_cache
 
-        # Specs are by default not assumed to be normal or concrete.
-        self._normal = False
-        self._concrete = False
+        self._prefix = getattr(other, '_prefix', None)
+
+        # Specs are by default not assumed to be normal, but in some
+        # cases we've read them from a file want to assume normal.
+        # This allows us to manipulate specs that Spack doesn't have
+        # package.py files for.
+        self._normal = kwargs.get('normal', False)
```

Comment:
@becker33 I did it in 1fdcb909d89a320ce10bb290f66cdf3a24855de9 to clean up a bit the class. Reason:

>These two keyword arguments where only used in one test module to force a Spec to 'appear' concrete. I suspect they are just a leftover from another refactoring, as now there's the private method '_mark_concrete' that does essentially the same job. Removed them to reduce a bit the
clutter in Spec.